cranelift/codegen/src/egraph/cost.rs

```diff
@@ -30,24 +30,86 @@ use crate::ir::Opcode;
 /// `finite()` method.) An infinite cost is used to represent a value
 /// that cannot be computed, or otherwise serve as a sentinel when
 /// performing search for the lowest-cost representation of a value.
-#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord)]
-pub(crate) struct Cost(u32);
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub(crate) struct Cost {
+    opcode_cost: u32,
+    depth: u32,
+}
```

Comment:
Drive-by comment here (sorry!): it might be nice to pack this into a single `u32`; CLIF data structures in general are pretty sensitive to size in memory, or at least that's been my experience. Maybe 8 bits for depth and 24 for cost (since cost has a bit more range with deep trees, potential loop nest multipliers at some point, etc)?